pages/api/openPR.js

```diff
@@ -15,40 +15,178 @@ const GITHUB_BRANCH = process.env.GITHUB_BRANCH
+    "-" +
+    (Math.random() * 10 ** 16).toString(36);
+  return checkoutBranch;
+}
+
+// Get SDG relevance info
+function getSDGRelevanceInfo(values, sdgNumber, evidenceText) {
+  //Loop through SDG array and parse SDG information
+  for (let i = 0; i < values.SDGs.length; i++) {
+    sdgNumber = parseInt(values.SDGs[i]);
+    evidenceText = "evidenceText".concat(sdgNumber);
+
+    values.SDGs[i] = {
+      SDGNumber: sdgNumber,
+      evidenceText: values[evidenceText],
+    };
+
+    delete values[evidenceText];
+  }
+  return values;
+}
+
+// Order fields e.g organizations
+function orderFields(values) {
+  let name, website, org_type, contact_name, contact_email;
+  // Order all entries for organizations
+  for (let i = 0; i < values.organizations.length; i++) {
+    name = values.organizations[i].name;
+    website = values.organizations[i].website;
+    org_type = values.organizations[i].org_type;
+    contact_name = values.organizations[i].contact_name;
+    contact_email = values.organizations[i].contact_email;
+
+    values.organizations[i] = {
+      name: name,
+      website: website,
+      org_type: org_type,
+      contact_name: contact_name,
+      contact_email: contact_email,
+    };
+  }
+  return values;
+}
+
+// Return multiple submission files including corresponding filepaths
+function getSubmissionFiles(values, nomineeJSON) {
+  let name,
+    nomineePath,
+    dpgPath,
+    stage,
+    data,
+    files = {};
+
+  stage = values.stage;
+
+  // Delete multiple DPG nomination fields
+  [
+    "aliases",
+    "description",
+    "website",
+    "type",
+    "SDGs",
+    "license",
+    "organizations",
+    "stage",
+    "repositoryURL",
+  ].forEach((e) => delete values[e]);
+
+  // Convert JavaScript submission sorted object into JSON string and add newline at EOF
+  data = JSON.stringify(values, null, 2).concat("\n");
+
+  // Parse project names using DPG naming convention for filenames
+  name = parseProjectName(values);
+
+  // Add nominee submission to nominee directory
+  nomineePath = "nominees/" + `${name}`;
+
+  // Add DPG submission to screening directory
+  dpgPath = "screening/" + `${name}`;
+
+  stage === "nominee"
+    ? Object.entries(nomineeJSON).forEach(
+        () =>
+          (files = {
+            [nomineePath]: {
+              content: nomineeJSON,
+              encoding: "utf-8",
+            },
+          })
+      )
+    : Object.entries(nomineeJSON).forEach(
+        () =>
+          (files = {
+            [nomineePath]: {
+              content: nomineeJSON,
+              encoding: "utf-8",
+            },
+            [dpgPath]: {
+              content: data,
+              encoding: "utf-8",
+            },
+          })
+      );
+
+  return files;
+}
+
+// Nominee processing before opening pull request
+function nomineeSubmission(values, sortedSubmission) {
+  // Sort entries by iterating through object and unpacking entries
+  Object.entries(values).forEach(
```

Comment:
Same observation as above, iterating through the object is not needed.

Reply:
Resolved in commit dd5cfcf.